sunbeam_migrate/handlers/neutron/floating_ip.py

```diff
@@ -0,0 +1,171 @@
+# SPDX-FileCopyrightText: 2025 - Canonical Ltd
+# SPDX-License-Identifier: Apache-2.0
+
+import ipaddress
+
+from sunbeam_migrate import config, exception
+from sunbeam_migrate.handlers import base
+
+CONF = config.get_config()
+
+
+class FloatingIPHandler(base.BaseMigrationHandler):
+    """Handle Neutron floating IP migrations."""
+
+    def get_service_type(self) -> str:
+        """Get the service type for this type of resource."""
+        return "neutron"
+
+    def get_supported_resource_filters(self) -> list[str]:
+        """Get a list of supported resource filters.
+
+        These filters can be specified when initiating batch migrations.
+        """
+        return ["owner_id"]
+
+    def get_associated_resource_types(self) -> list[str]:
+        """Get a list of associated resource types.
+
+        Associated resources must be migrated first.
+        """
+        return ["network", "subnet"]
+
+    def get_associated_resources(self, resource_id: str) -> list[tuple[str, str]]:
+        """Return the network and subnet this floating IP depends on."""
+        source_fip = self._source_session.network.get_ip(resource_id)
+        if not source_fip:
+            raise exception.NotFound(f"Floating IP not found: {resource_id}")
+
+        associated_resources = []
+        floating_network_id = source_fip.floating_network_id
+        if floating_network_id:
+            associated_resources.append(("network", floating_network_id))
+
+        subnet_ids = set()
+        if getattr(source_fip, "subnet_id", None):
```

Comment:
Let's avoid unnecessary getattr calls. Those fields are part of the FIP structure and are guaranteed to be accessible.